tendrl/gluster_integration/manager/manager.py

```diff
@@ -149,15 +151,79 @@ def on_pull(self, raw_data, cluster_id):
                     self.persister.update_volume(
```

Comment:
I feel if we merge this first and then rebase refactoring patch on top would be easier as its just additional code. But anything is fine.